elastica/modules/base_system.py

```diff
@@ -169,22 +170,13 @@ def finalize(self):
 
         # Toggle the finalize_flag
         self._finalize_flag = True
-        # sort _feature_group_synchronize so that _call_contacts is at the end
-        _call_contacts_index = []
-        for idx, feature in enumerate(self._feature_group_synchronize):
-            if feature.__name__ == "_call_contacts":
-                _call_contacts_index.append(idx)
-
-        # Move to the _call_contacts to the end of the _feature_group_synchronize list.
-        for index in _call_contacts_index:
-            self._feature_group_synchronize.append(
-                self._feature_group_synchronize.pop(index)
-            )
 
     def synchronize(self, time: float):
         # Collection call _feature_group_synchronize
         for feature in self._feature_group_synchronize:
             feature(time)
+        for feature in self._feature_group_synchronize_contact:
+            feature(time)
```

Comment:
Why do we need nested loops, and not add end of `self._feature_group_synchronize` list?

Reply:
yeah I am just saying why not do something like this somewhere in line 172. 

```python
self._feature_group_synchronize = self._self._feature_group_synchronize_connections + self._feature_group_synchronize_forcing  +  self._feature_group_synchronize_contact

```

We can explicitly set the order for forcing, connections and contact. Now connections are also part of `self._feature_group_synchronize`

Reply:
This is actually something I wanted to do, since we don't have a protocol in calling order of connections, forcing and contact before. We were only calling contact at the end.